pgpool/kubedb_client_builder.go

```diff
@@ -138,19 +141,51 @@ func (o *KubeDBClientBuilder) getBackendAuth() (string, string, error) {
 }
 
 func (o *KubeDBClientBuilder) getConnectionString() (string, error) {
-	user, pass, err := o.getBackendAuth()
-	if err != nil {
-		return "", err
-	}
-
 	if o.podName != "" {
 		o.url = o.getURL()
 	}
+	dnsName := o.url
 
 	if o.backendDBName == "" {
 		o.backendDBName = DefaultBackendDBName
 	}
-	//TODO ssl mode is disable now need to work on this after adding tls support
-	connector := fmt.Sprintf("user=%s password=%s host=%s port=%d connect_timeout=10 dbname=%s sslmode=%s", user, pass, o.url, DefaultPgpoolPort, o.backendDBName, TLSModeDisable)
-	return connector, nil
+
+	user, pass, err := o.getBackendAuth()
+	if err != nil {
+		return "", fmt.Errorf("DB basic auth is not found for backend PostgreSQL %v/%v", o.pgpool.Namespace, o.pgpool.Name)
```

Comment:
If client want to authenticate using certificate, then password has no significance. In this case user is allowed to not have auth secret. So you should only reject if bot (username,password) and (client cert) is not provided.